modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/util/TypeUtils.java

```diff
@@ -180,14 +180,22 @@ private static RelDataType createRowType(IgniteTypeFactory typeFactory, List<Rel
     }
 
     /** */
-    public static RelDataType sqlType(IgniteTypeFactory typeFactory, Class<?> cls, int precision, int scale) {
+    public static RelDataType sqlType(
+        IgniteTypeFactory typeFactory,
+        Class<?> cls,
+        int precision,
+        int scale,
+        boolean nullability
+    ) {
         RelDataType javaType = typeFactory.createJavaType(cls);
 
         if (javaType.getSqlTypeName().allowsPrecScale(true, true) &&
-            (precision != RelDataType.PRECISION_NOT_SPECIFIED || scale != RelDataType.SCALE_NOT_SPECIFIED))
-            return typeFactory.createSqlType(javaType.getSqlTypeName(), precision, scale);
+            (precision != RelDataType.PRECISION_NOT_SPECIFIED || scale != RelDataType.SCALE_NOT_SPECIFIED)) {
+            return typeFactory.createTypeWithNullability(
+                typeFactory.createSqlType(javaType.getSqlTypeName(), precision, scale), nullability);
+        }
 
-        return sqlType(typeFactory, javaType);
+        return typeFactory.createTypeWithNullability(sqlType(typeFactory, javaType), nullability);
```

Comment:
Do we have any test for this?

Reply:
`alterTableAddNotNullColumn` for example. There was another error message in this test, because null values were allowed for type and value was checked later for nullability (during insert). Now type doesn't allow null values and error message changed in test.